wp-content/civi-extensions/goonjcustom/CRM/Goonjcustom/CivirulesAction/GenerateQrCodeForContact.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+require_once __DIR__ . '/../../vendor/autoload.php';
+
+use chillerlan\QRCode\QRCode;
+use chillerlan\QRCode\QROptions;
+
+/**
+ * @file
+ */
+
+/**
+ *
+ */
+class CRM_Goonjcustom_CivirulesAction_GenerateQrCodeForContact extends CRM_Civirules_Action
+{
+		/**
+		 * Method processAction to execute the action.
+		 *
+		 * @param CRM_Civirules_TriggerData_TriggerData $triggerData
+		 *
+		 * @access public
+		 */
+		public function processAction(CRM_Civirules_TriggerData_TriggerData $triggerData)
+		{
+				$contactId = $triggerData->getContactId();
+
+				try {
+						// Todo - Switch to apiv4
+
+						$contact = \Civi\Api4\Contact::get(true)
+								->addWhere('id', '=', $contactId)
+								->setLimit(25)
+								->execute();
```

Comment:
I don't think we need to fetch the contact details at all! At least based on the current logic. The only thing needed is the `$contactId` which we already have.